### PR TITLE
Add visual indicator for piece purchases

### PR DESCRIPTION
**Pull request by esm2000**
## Summary
- Records spawned pieces in `latest_movement` with a `spawned: true` flag so purchases are visible in game history
- Adds a gold flash overlay + glow animation on purchased pieces so players can see when the CPU buys a piece
- Responsive: overlay matches piece dimensions on both desktop and mobile

## Test plan
- [x] CPU purchases a piece and it flashes gold on the board
- [x] Gold flash is visible and noticeable against the board background
- [x] Flash works on both desktop and mobile viewports
- [x] Regular piece moves do not trigger the gold flash
- [x] `latest_movement` correctly records spawned pieces with `spawned: true`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by esm2000** [comment]
@codex @gemini-code-assist @claude review

**Comment by esm2000** [comment]
Thanks for the thorough review! Addressed in `9497512d` and `23803d99`:

- **#1 Type mismatch**: Resolved by moving spawns to a separate `latest_spawns` field — no `spawned` key in `MovedPiece` needed. Added `latest_spawns: LatestMovement` to `GameState`.
- **#2 Stale docstring**: Updated to reflect both `latest_movement` and `latest_spawns`.
- **#3 Unit tests**: Added 6 unit tests covering regular moves, spawn-only turns, captures, mixed moves+spawns, and the critical case where spawn-only turns preserve `latest_movement`.
- **#4 Animation re-trigger**: Skipping — edge case, CSS animation naturally completes on first play.
- **#5 Overlay in DOM**: Skipping — cosmetic, `pointer-events: none` protects interactions.
- **#6 Filter compatibility**: Noted, all target browsers support chained filter functions.
- **#7 Hardcoded dimensions**: Noted for future, not an issue today.

File: backend/src/utils/game_state.py [hunk 1]
```diff
@@ -138,11 +138,18 @@ def is_captured_or_spawned(moved_pieces_entry: MovedPiece) -> bool:
...
+        if spawned_pieces:
+            record = record + [{"piece": entry["piece"], "side": entry["side"],
+                                "previous_position": entry["previous_position"],
+                                "current_position": entry["current_position"],
+                                "spawned": True} for entry in spawned_pieces]
         new_game_state["latest_movement"] = {
             "turn_count": new_game_state["turn_count"],
-            "record": filtered_moved_pieces
+            "record": record
         }
```
[the remark this answers is not in the document]
Comment (esm2000) [reply backend/src/utils/game_state.py]: Fixed in `9497512d`: refactored to use a single loop for spawned pieces and separated them into `latest_spawns` field, which also simplified the iteration.

File: backend/src/utils/game_state.py [hunk 2]
```diff
@@ -138,11 +138,18 @@ def is_captured_or_spawned(moved_pieces_entry: MovedPiece) -> bool:
         return moved_pieces_entry["previous_position"][0] is None \
         or  moved_pieces_entry["current_position"][0] is None
     filtered_moved_pieces = [entry for entry in moved_pieces if not is_captured_or_spawned(entry)]
-
-    if filtered_moved_pieces:
+    spawned_pieces = [entry for entry in moved_pieces if entry["previous_position"][0] is None and entry["current_position"][0] is not None]
+
+    if filtered_moved_pieces or spawned_pieces:
```
[the remark this answers is not in the document]
Comment (esm2000) [reply backend/src/utils/game_state.py]: Fixed in `9497512d`: spawns are now stored in a separate `latest_spawns` field, so `latest_movement` is never overwritten on spawn-only turns. Added unit test `test_spawn_only_does_not_overwrite_latest_movement` to verify.

**Comment by esm2000** [comment]
Also addressed #4 and #5 in `08a69b0d`:

- **#4 Animation re-trigger**: Added local `showPurchaseFlash` state in Piece.js. Initializes from `props.purchased`, clears on `onAnimationEnd` — so the flash only plays once even if the component remounts during the same turn.
- **#5 Overlay in DOM**: The overlay div is now removed from the DOM entirely when `onAnimationEnd` fires (sets `showPurchaseFlash` to false, which unmounts the overlay).

All actionable review comments are now addressed. #6 (filter compatibility) and #7 (hardcoded dimensions) are noted but not code changes.

**Comment by esm2000** [comment]
Also addressed #7 in `1b9f5ad2`: overlay now inherits dimensions from the piece's CSS class (`regular_piece`, `dragon_piece`, etc.) instead of hardcoding sizes. Removed the mobile override too since it's no longer needed.
